### ISSUE TITLE
When opening a post via a URL link, post doesn't get added to Post Tracker

### ISSUE THREAD

**Issue by boscojwho** [mlemgroup/mlem#635]
**Describe the bug**
Opening a text link to a Lemmy post (lemmy.world/post/[postId]) doesn't cause that post to get marked as read because it isn't added to Post Tracker.
<img src="https://github.com/mlemgroup/mlem/assets/2549615/61b3a98b-9370-44eb-bb64-a84fd71a1593" width="360">

**Expected Behavior**
- Post gets added to post tracker.
- Post gets marked as read (and doesn't crash on dev builds).

**Actual Behavior**
- Crashes on dev builds.
- Error haptics plays on other builds because post failed to be marked as read. 

**To Reproduce**
1. Open post with text link to another Lemmy post.
2. Tap on link.
3. Resolver will open post in app, but fail to mark it as read because `post.uid` doesn't exist in Post Tracker. (This is a crash on assertionFailure in dev mode).

**Mlem Information:**
- Latest `/dev` commit.

**Comment by boscojwho** [comment]
Example: lookup: https://lemmy.ml/post/5323962 (original: https://lemmy.ml/post/5323962)
- `PostTracker.id` set is empty, but `items` is not (and contains the correct post).
- But, `PostTracker`'s guard tests against `post.uid` and not `post.apId`.

```
(lldb) po self.items.first?.post.id
▿ Optional<Int>
  - some : 5517523
(lldb) po self.items.first?.post.apId
▿ Optional<String>
  - some : "https://lemmy.ml/post/5323962"
(lldb) po self.items.first?.uid
▿ Optional<ContentModelIdentifier>
  ▿ some : ContentModelIdentifier
    - contentType : Mlem.ContentType.post
    - contentId : 5517523
(lldb) po self.items.first?.post.name
▿ Optional<String>
  - some : "The Age of the Grift Shift"
```

**Comment by boscojwho** [comment]
Ah, `HandleLemmyLinkResolution` resolves the post object for us, but that code path doesn't trigger `PostTracker.add(...)`, which is what inserts ids into the tracker in `deduped(...)`.

**Comment by EricBAndrews** [comment]
Fixed in #669
